frontend/src/features/gallery/CurrentImage.tsx

```diff
@@ -0,0 +1,155 @@
+import {
+    Box,
+    Center,
+    Flex,
+    IconButton,
+    Image,
+    Link,
+    Tooltip,
+    useColorModeValue,
+    VStack,
+} from '@chakra-ui/react';
+import { FaCopy, FaRecycle } from 'react-icons/fa';
+import { RiBracesFill } from 'react-icons/ri';
+import { GiResize } from 'react-icons/gi';
+import { IoOpen } from 'react-icons/io5';
+import { MdFaceRetouchingNatural } from 'react-icons/md';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { RootState } from '../../app/store';
+import { setAllParameters, setInitialImagePath } from '../sd/sdSlice';
+import { useState } from 'react';
+import ImageMetadataViewer from './ImageMetadataViewer';
+import DeleteImageModalButton from './DeleteImageModalButton';
+
+const height = 'calc(100vh - 216px)';
+
+const CurrentImage = () => {
+    const { currentImageUuid, images, intermediateImage } = useAppSelector(
+        (state: RootState) => state.gallery
+    );
+
+    const { isGFPGANAvailable, isESRGANAvailable } = useAppSelector(
+        (state: RootState) => state.system
+    );
+    const dispatch = useAppDispatch();
+
+    const [shouldShowImageDetails, setShouldShowImageDetails] =
+        useState<boolean>(false);
+
+    const imageToDisplay =
+        intermediateImage ||
+        images.find((image) => image.uuid === currentImageUuid);
+
+    const bgColor = useColorModeValue(
+        'rgba(255, 255, 255, 0.85)',
+        'rgba(0, 0, 0, 0.8)'
+    );
+
+    return (
+        <Center height={height}>
+            {imageToDisplay && (
+                <Flex gap={2}>
+                    <Box position={'relative'}>
+                        <Image
+                            height={height}
+                            src={imageToDisplay.url}
+                            fit='contain'
+                            // minWidth={512}
+                            // minHeight={512}
+                        />
```

Comment:
This `height` value (`calc(100vh - 216px)`) will stretch images beyond their original size, which isn't optimal to look at (and could make it harder to distinguish upscaled from non-upscaled images?)

Maybe instead of setting the `height` explicitly, constrain the image dimensions with `max-(width/height)` and then let `object-fit: contain` do the rest of the work:
```suggestion
                        <Image
                            src={imageToDisplay.url}
                            fit='contain'
                            maxWidth='100%'
                            maxHeight='100%'
                        />
```

Reply:
I had some issues with containing the image related to the layout being done with CSS grid using max-width/max-height and object-fit and ended up just doing what I did to get around it. IIRC image overflowed the grid without manually setting its height. I'll revisit this as I agree it makes for bad display of the images.